app/plugin_hooks_test.go

```diff
@@ -622,7 +622,10 @@ func TestHookFileWillBeUploaded(t *testing.T) {
 				p.API.LogDebug(buf.String())
 
 				outbuf := bytes.NewBufferString("changedtext")
-				io.Copy(output, outbuf)
+				_, err := io.Copy(output, outbuf)
+				if err != nil {
+					panic("io.Copy failed: " + err.Error())
```

Comment:
Why not return `err` and then it can be checked in the [assert after the `UploadFiles`](https://github.com/mattermost/mattermost-server/blob/e0c443105d3fbb8cd5112cb26e1b40c6e18f5e8d/app/plugin_hooks_test.go#L649)? 🤔

Reply:
Returning the error will send it via an RPC before we have the next time of catching it, we want to print it out immediately. We may want to capture `n` and print it out too, just in case.

Reply:
Ah, gotcha! 👍